File: src/main/java/com/tms/csp/parse/Head.java
package com.tms.csp.parse;

import com.tms.csp.ast.PLConstants;

public class Head implements PLConstants {


    public static boolean isNegated(int lit) {
        return lit < 1;
    }

    public static int getVarId(int lit) {
        return Math.abs(lit);
    }

    public static boolean getSign(int lit) {
        return lit >= 0;
    }

    public static boolean getSign(String head) {
        return !isNegated(head);
    }

    public static String getCode(int head) {
        return Math.abs(head) + "";
    }

    public static boolean isNegated(String head) {
        return head.charAt(0) == BANG;
    }

    public static String getVarCode(String head) {
        if (isNegated(head)) {
            return head.substring(1);
        } else {
            return head;
        }
    }

    public static boolean isConstantTrue(String head) {
        return head.equals(TRUE_TOKEN);
    }

    public static boolean isConstantFalse(String head) {
        return head.equals(FALSE_TOKEN);
    }


}

File: src/main/java/com/tms/csp/parse/Head.kt
package com.tms.csp.parse

import com.tms.csp.ast.PLConstants
import com.tms.csp.ast.PLConstants.*

class Head : PLConstants {
    companion object {


        fun isNegated(lit: Int): Boolean {
            return lit < 1
        }

        fun getVarId(lit: Int): Int {
            return Math.abs(lit)
        }

        fun getSign(lit: Int): Boolean {
            return lit >= 0
        }

        fun getSign(head: String): Boolean {
            return !isNegated(head)
        }

        fun getCode(head: Int): String {
            return Math.abs(head).toString() + ""
        }

        fun isNegated(head: String): Boolean {
            return head[0] == BANG
        }

        fun getVarCode(head: String): String {
            return if (isNegated(head)) {
                head.substring(1)
            } else {
                head
            }
        }

        fun isConstantTrue(head: String): Boolean {
            return head == TRUE_TOKEN
        }

        fun isConstantFalse(head: String): Boolean {
            return head == FALSE_TOKEN
        }
    }


}
